src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -360,7 +360,21 @@ void NonClientIslandWindow::_OnMaximizeChange() noexcept
 //   sizes of our child XAML Islands to match our new sizing.
 void NonClientIslandWindow::_UpdateIslandPosition(const UINT windowWidth, const UINT windowHeight)
 {
-    const auto topBorderHeight = Utils::ClampToShortMax(_GetTopBorderHeight(), 0);
+    const auto originalTopHeight = _GetTopBorderHeight();
+    // GH#7422
+    // !! BODGY !!

```

Comment:
I was about to say "did you include my magic keyword of `BODGY`"? 

Yes. Yes you did.